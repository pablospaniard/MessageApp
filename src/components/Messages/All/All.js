import React, { Component } from 'react'
import { Table } from 'reactstrap'

import { NoMessages } from '../../Messages'

class All extends Component {
  render() {
    return (
      <Table>
        <thead>
          <tr>
            <th>#</th>
            <th>Type</th>
            <th>Recipient</th>
            <th>Originator</th>
            <th>Message</th>
            <th>Status</th>
            <th>Date</th>
            <th />
          </tr>
        </thead>
        <tbody>
          <tr>
            <th scope="row">1</th>
            <td>></td>
            <td>+34672972125</td>
            <td>Pavel</td>
            <td>Text</td>
            <td>delivered</td>
            <td>09:52</td>
          </tr>
        </tbody>
      </Table>
    )
  }
}

export default All
