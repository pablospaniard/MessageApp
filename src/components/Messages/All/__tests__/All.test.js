import React from 'react'
import { render, cleanup } from 'react-testing-library'
import All from '../All'

const messages = {
  items: [
    {
      id: 'e8077d803532c0b5937c639b60216938',
      href:
        'https://rest.messagebird.com/messages/e8077d803532c0b5937c639b60216938',
      direction: 'mt',
      type: 'sms',
      originator: 'MessageBird',
      body: 'The message to be sent',
      reference: 'the-client-reference',
      validity: null,
      gateway: 240,
      typeDetails: {},
      datacoding: 'plain',
      mclass: 1,
      scheduledDatetime: null,
      createdDatetime: '2016-04-29T09:42:26+00:00',
      recipients: {
        totalCount: 1,
        totalSentCount: 1,
        totalDeliveredCount: 0,
        totalDeliveryFailedCount: 0,
        items: [
          {
            recipient: 31612345678,
            status: 'sent',
            statusDatetime: '2016-04-29T09:42:26+00:00'
          }
        ]
      }
    }
  ]
}

describe('All should render properly', () => {
  afterEach(cleanup)

  it('should match snapshot', () => {
    const { container } = render(<All messages={messages} />)
    expect(container.firstChild).toMatchSnapshot('All_snapshot_1')
  })

  it('should show spinner if loading is true', () => {
    const { getByTestId } = render(<All messages={messages} loading />)
    expect(getByTestId('spinner'))
  })
})
