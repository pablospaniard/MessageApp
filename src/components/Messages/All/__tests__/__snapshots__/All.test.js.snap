// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All_snapshot_1 1`] = `
<table
  class="table"
>
  <thead>
    <tr>
      <th>
        Type
      </th>
      <th>
        Recipient
      </th>
      <th>
        Originator
      </th>
      <th>
        Message
      </th>
      <th>
        Status
      </th>
      <th>
        Date
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        class="sc-EHOje cRULWI"
      >
        Outgoing
      </td>
      <td
        class="sc-EHOje cRULWI"
      >
        31612345678
      </td>
      <td
        class="sc-EHOje cRULWI"
      >
        MessageBird
      </td>
      <td
        class="sc-EHOje cRULWI"
      >
        The message to be sent
      </td>
      <td
        class="sc-EHOje cRULWI"
      >
        sent
      </td>
      <td
        class="sc-EHOje cRULWI"
      >
        29-04-16 11:42
      </td>
    </tr>
  </tbody>
</table>
`;
